### ISSUE TITLE
Updating the publish date should update the post

### ISSUE THREAD

**Issue by oguzkocer** [wordpress-mobile/WordPress-Android#5927]
![screenshot_20170523-1500591](https://cloud.githubusercontent.com/assets/662023/26400113/2a60c670-4088-11e7-95ac-19a4d1c0c31a.png)

**Comment by oguzkocer** [comment]
I've updated the title of this issue as it turns out the issue is not what it initially seemed. Here are a few paths that should be checked when this is worked on:

```
1. Create a local draft
2. Go into edit and change the publish date
3. At this point the “Update” button should change to “Schedule” but it doesn’t look like it does. Also, updating the post doesn’t seem to schedule it which leads to showing the “Publish” button instead of the “Sync” since it’s not really scheduled.
```

```
1. Start a new post using WPAndroid alpha-58.
2. Leave the writing area without making changes to the publish date.
3. Find the post in the Blog Posts list.
4. Change the publish date to something the future.
5. Either tap the back arrow (top left) or the “Update” button (top right). (At this point it is no longer a local draft.)
```

This is also somewhat relevant to #5919.

**Comment by oguzkocer** [comment]
Fixed via #6122 & #6185
